storage/src/node/mod.rs

```diff
@@ -272,24 +267,27 @@ impl Node {
 
                 // encode the children
                 if childcount == BranchNode::MAX_CHILDREN {
-                    for (_, child) in child_iter {
-                        if let Child::AddressWithHash(address, hash) = child {
+                    for child in b.children.iter() {
+                        if let Some(Child::AddressWithHash(address, hash)) = child {
                             encoded.extend_from_slice(&address.get().to_ne_bytes());
-                            encoded.extend_from_slice(&hash.serialized_bytes());
+                            hash.write_to_vec(encoded);
                         } else {
                             panic!(
                                 "attempt to serialize to persist a branch with a child that is not an AddressWithHash"
                             );
                         }
                     }
                 } else {
-                    for (position, child) in child_iter {
+                    for (position, child) in b.children.iter().enumerate() {
```

Comment:
Also surprised on this one. Are you sure that unrolling the filter from the iterator is faster than checking it yourself?